rfcs/theme-shape.md

```diff
@@ -0,0 +1,228 @@
+# RFC: Update theme shape
+
+@miroslavstastny, @layershifter
+
+_This RFC updates [theme-tokens.md](./theme-tokens.md) RFC. Once this on is approved, we should merge the two RFCs together._
+
+## Summary
+
+As described in [theme-tokens.md](./theme-tokens.md) the current theme consists of global and alias tokens.
+All tokens are injected into DOM as CSS variables, resulting in 1200 CSS variables. With this approach, when a theme switches, there is no need to re-compute any component styles, by just setting the new token values to the CSS variables, all components change their appearance accordingly.
+
+However, 1200 CSS variables in DOM affects browser performance. This RFC proposes ways how to improve theme performance.
+
+## <a name="performance-analysis"></a>Performance analysis
+
+The current theme consists of ~1200 tokens which are injected into DOM as CSS variables. It impacts not only Javascript performance but also browser performance in styles computation phase.
+
+To see the performance impact we render 20 `FluentProvider` components side by side (**each** injecting a class with 1200 CSS variables) - mount and unmount 10 times and measure the performance in Chrome profiler with 6x CPU slowdown. After each mount, a CSS attribute is accessed to force reflow in order to be able to measure the rendering performance.
+
+### Master
+
+Current implementation in `@fluentui/react-theme@9.0.0-alpha.22`.
+
+- 1200 CSS variables in a theme = ~650 global tokens + ~550 alias tokens
+- alias token implemented as a "reference" to a global token: `--alias-token: var(--global-token)`.
+- **Each reflow takes ~150ms (1500 ms total for the 10 mounts)**
+
+![Flamechart - Master](theme-shape/perf-master.png)
+
+### Inline values to alias tokens.
+
+- Still the same 1200 CSS variables (650 global + 550 alias)
+- Instead of "referencing" global tokens in the alias tokens, inline the values: `--alias-token: #fff`.
+- **Each reflow takes ~90ms (1000 ms total for the 10 mounts)**
+
+![Flamechart - Inline values to alias tokens](theme-shape/perf-inline-alias.png)
+
+### Reduce number of tokens (CSS variables)
+
+- Removed global color tokens, number of CSS variables reduced to 550.
+- **Each reflow takes ~50ms (600 ms total for the 10 mounts)**
+
+![Flamechart - Reduce number of tokens](theme-shape/perf-no-global.png)
+
+### Remove even more tokens
```

Comment:
It seems like it will be far easier to add more tokens in the future, rather than remove them. As long as this is true, given the perf impact, I'd recommend we reduce our # of tokens now and add tokens based on need later